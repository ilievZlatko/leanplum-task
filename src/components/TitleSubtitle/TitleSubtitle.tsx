import React from 'react'

interface Props {
	title: string | React.ReactNode
	subtitle: string | React.ReactNode
}

const TitleSubtitle: React.FC<Props> = ({ title, subtitle, ...otherProps }) => {
	return (
		<div {...otherProps}>
			<h2 data-test='title' className='title'>
				{title}
			</h2>
			<span data-test='subtitle' className='subtitle'>
				{subtitle}
			</span>
		</div>
	)
}

export default TitleSubtitle
