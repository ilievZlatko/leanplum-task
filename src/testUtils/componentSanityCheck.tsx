import React from 'react'
import { shallow } from 'enzyme'

export const componentSanityCheck = (
	label: string,
	Element: React.ReactElement,
) =>
	describe(`Sanity checks for ${label}`, () => {
		it('is defined', () => {
			expect(Element).toBeDefined()
		})

		it('renders the element', () => {
			const wrapper = shallow(Element)
			const expected = false
			const actual = wrapper.isEmptyRender()
			expect(expected).toEqual(actual)
		})

		it('matches the snapshot', () => {
			const wrapper = shallow(Element)
			const expected = wrapper.html()
			expect(expected).toMatchSnapshot()
		})
	})
