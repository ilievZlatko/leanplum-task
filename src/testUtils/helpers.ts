import { ShallowWrapper, ReactWrapper } from 'enzyme'

export const findByTestAttr = (
	component: ShallowWrapper | ReactWrapper | any,
	attr: string,
): ShallowWrapper | ReactWrapper => {
	return component.find(`[data-test='${attr}']`)
}
