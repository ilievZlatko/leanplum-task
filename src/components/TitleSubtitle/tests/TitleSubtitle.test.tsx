import React from 'react'
import { shallow } from 'enzyme'

import TitleSubtitle from '../TitleSubtitle'
import { componentSanityCheck } from '../../../testUtils/componentSanityCheck'
import { findByTestAttr } from '../../../testUtils/helpers'

describe('TitleSubtitle', () => {
	componentSanityCheck(
		'TitleSubtitle',
		<TitleSubtitle title='test title' subtitle='test content' />,
	)

	it('Should accept title and subtitle of type ReactNode', (done: jest.DoneCallback) => {
		const wrapper = shallow(
			<TitleSubtitle
				title={<span>Title</span>}
				subtitle={<span>Subtitle</span>}
			/>,
		)

		expect(wrapper.html()).toContain('<span>Title</span>')
		expect(wrapper.html()).toContain('<span>Subtitle</span>')
		done()
	})

	it('Should accept title and subtitle of type string', (done: jest.DoneCallback) => {
		const wrapper = shallow(
			<TitleSubtitle title='Info title' subtitle='Info subtitle' />,
		)

		const title = findByTestAttr(wrapper, 'title')
		const subtitle = findByTestAttr(wrapper, 'subtitle')

		expect(title.text()).toEqual('Info title')
		expect(subtitle.text()).toEqual('Info subtitle')
		done()
	})
})
