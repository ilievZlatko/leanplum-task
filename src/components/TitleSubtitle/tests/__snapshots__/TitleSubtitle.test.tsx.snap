// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleSubtitle Sanity checks for TitleSubtitle matches the snapshot 1`] = `"<div><h2 data-test=\\"title\\" class=\\"title\\">test title</h2><span data-test=\\"subtitle\\" class=\\"subtitle\\">test content</span></div>"`;
